From fc2de62496061b9d0bace02729765344bd939122 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Fri, 1 Aug 2014 12:00:05 +0100
Subject: [PATCH 1/2] sna/glyphs: Fallback to masked glyphs if rendering to a
 too wide dst

Fixes regression from

commit 79399ff9264ff23da0ab95131a67d2ac85651b3d
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Mon Mar 24 10:46:57 2014 +0000

    sna: Eliminate a few conditionals in glyph fast path

for large pixmaps (such as extended desktops on gen2/gen3).

Bugzilla: https://bugs.archlinux.org/task/40949
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
 src/sna/sna_glyphs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sna/sna_glyphs.c b/src/sna/sna_glyphs.c
index 0bb72d8..5d3bec2 100644
--- a/src/sna/sna_glyphs.c
+++ b/src/sna/sna_glyphs.c
@@ -800,7 +800,7 @@ glyphs0_to_dst(struct sna *sna,
 								   0, 0, 0, 0, 0, 0,
 								   0, 0,
 								   COMPOSITE_PARTIAL, &tmp))
-						goto next_glyph_N;
+						return false;
 
 					glyph_atlas = p->atlas;
 				}
@@ -892,7 +892,7 @@ next_glyph_N:
 							   0, 0, 0, 0, 0, 0,
 							   0, 0,
 							   COMPOSITE_PARTIAL, &tmp))
-					goto next_glyph_0;
+					return false;
 
 				glyph_atlas = p->atlas;
 			}
-- 
2.0.3

